platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/PomReader.java

```diff
@@ -728,6 +739,10 @@ private Map<String, String> parseProperties(Element parentElement) {
         return pomProperties;
     }
 
+    private static boolean hasUnresolvedProperty(String val) {
```

Comment:
🤔 Should we warn in this case?  This possible mistake remains completely invisible now.  Clearly attributing the problem to the particular dependency with the bad POM would (eventually) bring it to the attention of that lib's maintainers, who could correct their build in the next version.

I suppose if we warn we'd also need a flag to disable this warning (perhaps for specific POMs) so people could keep their build output clean.

Reply:
I can see why we would want to warn, but it also feels odd to me to warn when wee are following maven's default behavior.

If maven does this and does not warn, I don't think we should warn